plugins/modules/system/keyring.py

```diff
@@ -0,0 +1,282 @@
+description: >-
+  This module uses the L(keyring Python library, https://pypi.org/project/keyring/)
+  to set or delete passphrases for a given service and username from the OS' native keyring.
+requirements:
+  - keyring (Python library)
+  - gnome-keyring (application - required for headless Gnome keyring access)
+  - dbus-run-session (application - required for headless Gnome keyring access)
+options:
+  service:
+    description: The name of the service.
+    required: true
+    type: str
+  username:
+    description: The user belonging to the service.
+    required: true
+    type: str
+  user_password:
+    description: The password to set.
+    required: false
+    type: str
+    aliases:
+      - password
+  keyring_password:
+    description: Password to unlock keyring.
+    required: true
+    type: str
+  state:
+    description: Whether the password should exist.
+    required: false
+    default: present
+    type: str
+    choices:
+      - present
+      - absent
+"""
+
+EXAMPLES = r"""
+- name: test keyring module
+  hosts: localhost
+  connection: local
+  become: false
+  gather_facts: true
+  vars:
+    keyring_password: test
+    user_password: test12345
+  tasks:
+    - name: Set a password for test/test1
+      community.general.keyring:
+        service: test
+        username: test1
+        user_password: "{{ user_password }}"
+        keyring_password: "{{ keyring_password }}"
+      register: set_password
+      no_log: true
+
+    - name: Delete the password for test/test1
+      community.general.keyring:
+        service: test
+        username: test1
+        user_password: "{{ user_password }}"
+        keyring_password: "{{ keyring_password }}"
+        state: absent
+      register: delete_password
+      no_log: true
+"""
+
+try:
+    from shlex import quote
+except ImportError:
+    from pipes import quote
+from subprocess import Popen, PIPE
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+try:
+    import keyring
+
+    HAS_KEYRING = True
+except ImportError:
+    HAS_KEYRING = False
+    KEYRING_IMP_ERR = traceback.format_exc()
+
+
+def del_passphrase(module):
+    """
+    Attempt to delete passphrase in the keyring using a shell rather than the Python API.
+    This is primarily used for 'headless' hosts.
+    """
+    try:
+        keyring.delete_password(module.params["service"], module.params["username"])
+        return True
+    except keyring.errors.KeyringLocked as keyring_locked_err:  # pylint: disable=unused-variable
+        delete_argument = (
+            'echo "%s" | gnome-keyring-daemon --unlock\nkeyring del %s %s\n'
+            % (
+                quote(module.params["keyring_password"]),
+                quote(module.params["service"]),
+                quote(module.params["username"]),
+            )
+        )
+        alternate_delete = module.run_command(
```

Comment:
```suggestion
        dummy, dummy, rc = module.run_command(
```

Reply:
is `dummy` used inplace of something like `_`, to represent the unused elements of the tuple?

Reply:
Yes. (See https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/no-underscore-variable.html)